moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -1256,7 +1256,7 @@ template <typename T>
 bool parse(const YAML::Node& node, const std::string& key, T& storage, const T& default_value = T())
 {
   const YAML::Node& n = node[key];
-  bool valid = n;
+  bool valid = n.as<bool>();
```

Comment:
doesn't that break for a node containing `false` and a `default_value` of `true`

Reply:
```suggestion
  bool valid = n.isDefined();
```
but according to here https://yaml-cpp.docsforge.com/master/api/YAML/Node/operator%20bool/  operator bool exists. Why does it not compile on arch?

Reply:
And why does it say `boo` instead of `bool`?

Reply:
You are right! `IsDefined()` implements the correct semantics here. `as<bool>()` would attempt to read a boolean value.

> [`operator bool()`](https://yaml-cpp.docsforge.com/master/api/YAML/Node/operator%20bool/) exists. Why does it not compile on arch?
I fixed the commit.

The conversion operator is declared `explicit`. This wasn't the case in older versions used in Ubuntu.